src/cmd-buildextend-vmware

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# NOTE: PYTHONUNBUFFERED is set in cmdlib.sh for unbuffered output
+#
+# An operation that mutates a build by generating an ova
+
+import os
+import sys
+import json
+import shutil
+import argparse
+import tarfile
+
+sys.path.insert(0, '/usr/lib/coreos-assembler')
+from cmdlib import run_verbose, load_json, write_json, sha256sum_file
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--build", help="Build ID",
+                    required=False)
+args = parser.parse_args()
+
+# default to latest build if not specified
+if not args.build:
+    j = load_json('builds/builds.json')
+    args.build = j['builds'][0]
+
+builddir = os.path.join('builds', args.build)
+buildmeta_path = os.path.join(builddir, 'meta.json')
+buildmeta = load_json(buildmeta_path)
+base_name = buildmeta['name']
+vmware_prefix = f'{base_name}-{args.build}-vmware'
+ova_name = f'{vmware_prefix}.ova'
+
+try:
+    img_qemu = os.path.join(builddir, buildmeta['images']['qemu']['path'])
+except KeyError as e:
+    print("meta.json does not contain key .images.qemu.path")
+    sys.exit(1)
+
+tmpdir = 'tmp/buildpost-vmware'
+if os.path.isdir(tmpdir):
+    shutil.rmtree(tmpdir)
+os.mkdir(tmpdir)
+
+
+def generate_ovf_parameters(qemu, vmdk, cpu=2,
+                            memory=4096, system_type="vmx-07 vmx-08",
+                            os_type="other26xLinux64Guest", scsi="VirtualSCSI",
+                            network="VmxNet3"):
+    qemu_info = run_verbose(["qemu-img", "info", qemu,
+                             "--output", "json"], capture_output=True)
+    disk_size = json.loads(qemu_info.stdout)['virtual-size']
+    vmdk_size = str(os.stat(vmdk).st_blocks*512)
+
+    image = buildmeta['summary']
+    product = buildmeta['summary']
+    vendor = buildmeta['name']
+    version = buildmeta['ostree-version']
+
+    ova_params = {
+        'ovf_cpu_count':                    cpu,
+        'ovf_memory_mb':                    memory,
+        'vsphere_image_name':               image,
+        'vsphere_product_name':             product,
+        'vsphere_product_vendor_name':      vendor,
+        'vsphere_product_version':          version,
+        'vsphere_virtual_system_type':      system_type,
+        'vsphere_os_type':                  os_type,
+        'vsphere_scsi_controller_type':     scsi,
+        'vsphere_network_controller_type':  network,
+        'virtual_disk_size':                disk_size,
+        'vmdk_size':                        vmdk_size
+    }
+
+    return ova_params
+
+
+def generate_ova_from_template():
```

Comment:
In general I think it would be nice if we added a description to this function and some comments throughout the code. A strategy and some context of what makes up an OVA would be useful. 

Throughout the code we: 

- create vmware_qemu image with vmware platformid
- convert that to vmdk
- get the parameters for the ova
- apply the parameters to the template and write it out as xml
- add the xml file and vmdk to an ova (just a tar archive)
- update buildmeta

Reply:
I basically added all your comments!  Let me know if it looks okay